mlfinlab/ensemble/sb_bagging.py

```diff
@@ -0,0 +1,662 @@
+                             " if warm_start=False")
+
+        if not self.warm_start or not hasattr(self, 'estimators_'):
+            # Free allocated memory, if any
+            self.estimators_ = []
+            self.estimators_features_ = []
+
+        n_more_estimators = self.n_estimators - len(self.estimators_)
+
+        if n_more_estimators < 0:
+            raise ValueError('n_estimators=%d must be larger or equal to '
+                             'len(estimators_)=%d when warm_start==True'
+                             % (self.n_estimators, len(self.estimators_)))
+
+        elif n_more_estimators == 0:
+            warn("Warm-start fitting without increasing n_estimators does not "
+                 "fit new trees.")
+            return self
+
+        # Parallel loop
+        n_jobs, n_estimators, starts = _partition_estimators(n_more_estimators,
+                                                             self.n_jobs)
+        total_n_estimators = sum(n_estimators)
+
+        # Advance random state to state after training
+        # the first n_estimators
+        if self.warm_start and len(self.estimators_) > 0:
+            random_state.randint(MAX_INT, size=len(self.estimators_))
+
+        seeds = random_state.randint(MAX_INT, size=n_more_estimators)
+        self._seeds = seeds
+
+        # pylint: disable=C0330
+        all_results = Parallel(n_jobs=n_jobs, verbose=self.verbose,
+                               )(
+            delayed(_parallel_build_estimators)(
+                n_estimators[i],
+                self,
+                X,
+                y,
+                subsampled_ind_mat,
+                sample_weight,
+                seeds[starts[i]:starts[i + 1]],
+                total_n_estimators,
+                verbose=self.verbose)
+            for i in range(n_jobs))
+
+        # Reduce
+        self.estimators_ += list(itertools.chain.from_iterable(
+            t[0] for t in all_results))
+        self.estimators_features_ += list(itertools.chain.from_iterable(
+            t[1] for t in all_results))
+
+        if self.oob_score:
+            self._set_oob_score(X, y)
+
+        return self
+
+    def _get_estimators_indices(self):
+
+        # Get indicator matrix
+        subsampled_ind_mat = self.ind_mat[:, self.timestamp_int_index_mapping.loc[self.X_time_index]]
+
+        # Get drawn indices along both sample and feature axes
+        for seed in self._seeds:
+            # Operations accessing random_state must be performed identically
+            # to those in `_parallel_build_estimators()`
+            random_state = np.random.RandomState(seed)
+            feature_indices, sample_indices = _generate_bagging_indices(
+                random_state, self.bootstrap_features,
+                self.n_features_, self._max_features, self._max_samples, subsampled_ind_mat)
+
+            yield feature_indices, sample_indices
+
+    def estimators_samples_(self):
```

Comment:
This function deserves a better name like `get_estimators_samples` instead of just renaming the variable used in the original class `estimators_samples_`. Leaving it as is may lead to some confusion.